process/elasticProcessor.go

```diff
@@ -315,14 +315,35 @@ func (ei *elasticProcessor) RemoveMiniblocks(header coreData.HeaderHandler, body
 
 // RemoveTransactions will remove transaction that are in miniblock from the elasticsearch server
 func (ei *elasticProcessor) RemoveTransactions(header coreData.HeaderHandler, body *block.Body) error {
-	encodedTxsHashes := ei.transactionsProc.GetRewardsTxsHashesHexEncoded(header, body)
-	if len(encodedTxsHashes) == 0 {
+	encodedTxsHashes, encodedScrsHashes := ei.transactionsProc.GetHashesHexEncodedForRemove(header, body)
+
+	if len(encodedTxsHashes) != 0 {
+		err := ei.elasticClient.DoQueryRemove(
+			elasticIndexer.TransactionsIndex,
+			converters.PrepareHashesForQueryRemove(encodedTxsHashes),
+		)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Could create a func here (smth. like: `removeHashesIfNotEmpty`) with `index` + `[]string hashes` input and reuse below

Reply:
done.